File: scrape_august_10.js
const https = require('https');

function getWordleSolution(dateStr) {
    return new Promise((resolve, reject) => {
        const url = `https://www.nytimes.com/svc/wordle/v2/${dateStr}.json`;
        
        https.get(url, (res) => {
            let data = '';
            
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                try {
                    if (res.statusCode === 200) {
                        const jsonData = JSON.parse(data);
                        const solution = jsonData.solution;
                        
                        if (solution) {
                            console.log(`${dateStr} 的Wordle单词是：${solution.toUpperCase()}`);
                            resolve(solution.toUpperCase());
                        } else {
                            console.log(`${dateStr} 没有查到答案，返回内容：`, jsonData);
                            resolve(null);
                        }
                    } else {
                        console.log(`请求失败，状态码：${res.statusCode}`);
                        resolve(null);
                    }
                } catch (error) {
                    console.log('解析响应数据出错：', error);
                    resolve(null);
                }
            });
        }).on('error', (error) => {
            console.log('请求出错：', error);
            reject(error);
        });
    });
}

async function main() {
    try {
        const date = '2025-08-09';  // 改为8月9日
        const solution = await getWordleSolution(date);
        
        if (solution) {
            console.log(`\n成功获取到 ${date} 的Wordle答案：${solution}`);
        } else {
            console.log(`\n未能获取到 ${date} 的Wordle答案`);
        }
    } catch (error) {
        console.log('程序执行出错：', error);
    }
}

main(); 

File: scrape_august_10.py
import requests

def get_wordle_solution(date_str):
    """
    传入日期字符串（如'2025-07-31'），返回该日Wordle答案
    """
    url = f"https://www.nytimes.com/svc/wordle/v2/{date_str}.json"
    try:
        resp = requests.get(url, timeout=10)
        if resp.status_code == 200:
            data = resp.json()
            solution = data.get("solution")
            if solution:
                print(f"{date_str} 的Wordle单词是：{solution.upper()}")
                return solution.upper()
            else:
                print(f"{date_str} 没有查到答案，返回内容：{data}")
                return None
        else:
            print(f"请求失败，状态码：{resp.status_code}")
            return None
    except Exception as e:
        print("请求出错：", e)
        return None

if __name__ == "__main__":
    # 抓取8月10日的答案
    date = "2025-08-10"
    solution = get_wordle_solution(date)
    
    if solution:
        print(f"\n成功获取到 {date} 的Wordle答案：{solution}")
    else:
        print(f"\n未能获取到 {date} 的Wordle答案") 